osu.Game.Tests/Visual/UserInterface/TestSceneModPresetColumn.cs

```diff
@@ -261,6 +261,182 @@ public void TestDeleteFlow()
+
+            AddStep("right click first panel", () =>
+            {
+                var panel = this.ChildrenOfType<ModPresetPanel>().First();
+                previousMod = panel.Preset.Value.Mods.ToList();
+                InputManager.MoveMouseTo(panel);
+                InputManager.Click(MouseButton.Right);
+            });
+            AddUntilStep("wait for context menu", () => this.ChildrenOfType<OsuContextMenu>().Any());
+            AddStep("click edit", () =>
+            {
+                var editItem = this.ChildrenOfType<DrawableOsuMenuItem>().ElementAt(0);
+                InputManager.MoveMouseTo(editItem);
+                InputManager.Click(MouseButton.Left);
+            });
+
+            OsuPopover? popover = null;
+            AddUntilStep("wait for popover", () => (popover = this.ChildrenOfType<OsuPopover>().FirstOrDefault()) != null);
+            AddStep("click use current mods", () =>
+            {
+                InputManager.MoveMouseTo(popover.ChildrenOfType<ShearedButton>().ElementAt(0));
+                InputManager.Click(MouseButton.Left);
+            });
+            AddStep("attempt preset edit", () =>
+            {
+                InputManager.MoveMouseTo(popover.ChildrenOfType<ShearedButton>().ElementAt(1));
+                InputManager.Click(MouseButton.Left);
+            });
+            AddWaitStep("wait some", 3);
+            AddAssert("present mod not changed", () =>
+                new HashSet<Mod>(this.ChildrenOfType<ModPresetPanel>().First().Preset.Value.Mods).SetEquals(previousMod));
+
+            AddStep("select mods", () => SelectedMods.Value = mods);
+            AddStep("right click first panel", () =>
+            {
+                var panel = this.ChildrenOfType<ModPresetPanel>().First();
+                InputManager.MoveMouseTo(panel);
+                InputManager.Click(MouseButton.Right);
+            });
+
+            AddUntilStep("wait for context menu", () => this.ChildrenOfType<OsuContextMenu>().Any());
+            AddStep("click edit", () =>
+            {
+                var editItem = this.ChildrenOfType<DrawableOsuMenuItem>().ElementAt(0);
+                InputManager.MoveMouseTo(editItem);
+                InputManager.Click(MouseButton.Left);
+            });
+
+            AddUntilStep("wait for popover", () => (popover = this.ChildrenOfType<OsuPopover>().FirstOrDefault()) != null);
+            AddStep("click use current mods", () =>
+            {
+                InputManager.MoveMouseTo(popover.ChildrenOfType<ShearedButton>().ElementAt(0));
+                InputManager.Click(MouseButton.Left);
+            });
+            AddStep("attempt preset edit", () =>
+            {
+                InputManager.MoveMouseTo(popover.ChildrenOfType<ShearedButton>().ElementAt(1));
+                InputManager.Click(MouseButton.Left);
+            });
+            AddWaitStep("wait some", 3);
```

Comment:
Wait steps should not be used like this. There's no guarantee this is long enough.

Reply:
I will replace with AddUntilStep